File: app/src/main/java/com/androidexamples/app/activity/ActivityAnimationsActivity.java
package com.androidexamples.app.activity;

import android.content.Intent;
import android.support.v4.app.ActivityCompat;
import android.support.v4.app.ActivityOptionsCompat;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.view.Window;
import android.widget.ImageView;

import com.androidexamples.app.R;
import com.androidexamples.app.utils.Constants;

public class ActivityAnimationsActivity extends AppCompatActivity {

    // Views
    private ImageView imageView;

    // Another Objects
    private Intent intent;
    private Bundle bundle;
    private ActivityOptionsCompat activityOptionsCompat;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        getWindow().requestFeature(Window.FEATURE_CONTENT_TRANSITIONS);
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_animations_activity);
        loadComponents();
    }

    private void loadComponents() {
        setSupportActionBar((Toolbar)findViewById(R.id.top_toolbar));
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        imageView = (ImageView) findViewById(R.id.imageView);
        findViewById(R.id.button_fade).setOnClickListener(onFadeButtonClick());
        findViewById(R.id.button_slide_right).setOnClickListener(onSlideRightButtonClick());
        findViewById(R.id.button_slide_left).setOnClickListener(onSlideLeftButtonClick());
        findViewById(R.id.button_scale).setOnClickListener(onScaleButtonClick());
        findViewById(R.id.button_zoom).setOnClickListener(onZoomButtonClick());
        intent = new Intent(getBaseContext(), TestActivity.class);
        intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
    }

    private View.OnClickListener onZoomButtonClick() {
        return new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                ActivityOptionsCompat opts = ActivityOptionsCompat.makeCustomAnimation(getBaseContext(),
                        R.anim.zoom_enter, R.anim.zoom_enter);
                ActivityCompat.startActivity(getBaseContext(), intent, opts.toBundle());
            }
        };
    }

    private View.OnClickListener onScaleButtonClick() {
        return new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                bundle = ActivityOptionsCompat.makeScaleUpAnimation(v, 0, 0, v.getWidth(), v.getHeight()).toBundle();
                ActivityCompat.startActivity(getBaseContext(), intent, bundle);
            }
        };
    }

    private View.OnClickListener onSlideLeftButtonClick() {
        return new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                bundle = new Bundle();
                bundle.putInt(Constants.KEY_ANIMATION, Constants.ANIMATION_SLIDE_LEFT);
                intent.putExtras(bundle);

                activityOptionsCompat = ActivityOptionsCompat.makeCustomAnimation(getBaseContext(),
                        R.anim.slide_in_left, R.anim.slide_out_left);
                ActivityCompat.startActivity(getBaseContext(), intent, activityOptionsCompat.toBundle());
            }
        };
    }

    private View.OnClickListener onSlideRightButtonClick() {
        return new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                bundle = new Bundle();
                bundle.putInt(Constants.KEY_ANIMATION, Constants.ANIMATION_SLIDE_RIGHT);
                intent.putExtras(bundle);

                activityOptionsCompat = ActivityOptionsCompat.makeCustomAnimation(getBaseContext(),
                        R.anim.slide_in_right, R.anim.slide_out_right);
                ActivityCompat.startActivity(getBaseContext(), intent, activityOptionsCompat.toBundle());
            }
        };
    }

    private View.OnClickListener onFadeButtonClick() {
        return new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                bundle = new Bundle();
                bundle.putInt(Constants.KEY_ANIMATION, Constants.ANIMATION_FADE);
                intent.putExtras(bundle);

                activityOptionsCompat = ActivityOptionsCompat.makeCustomAnimation(getBaseContext(),
                        android.R.anim.fade_in, android.R.anim.fade_out);
                ActivityCompat.startActivity(getBaseContext(), intent, activityOptionsCompat.toBundle());
            }
        };
    }
}

File: app/src/main/java/com/androidexamples/app/flows/animations/activityAnimations/ActivityAnimationsActivity.kt
package com.androidexamples.app.flows.animations.activityAnimations

import android.content.Intent
import android.support.v4.app.ActivityCompat
import android.support.v4.app.ActivityOptionsCompat
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.Toolbar
import android.view.View
import android.view.Window
import android.widget.ImageView

import com.androidexamples.app.R
import com.androidexamples.app.utils.Constants

class ActivityAnimationsActivity : AppCompatActivity() {

    // Views
    private var imageView: ImageView? = null

    // Another Objects
    private var newIntent: Intent? = null
    private var bundle: Bundle? = null
    private var activityOptionsCompat: ActivityOptionsCompat? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        window.requestFeature(Window.FEATURE_CONTENT_TRANSITIONS)
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_animations_activity)
        loadComponents()
    }

    private fun loadComponents() {
        setSupportActionBar(findViewById<View>(R.id.top_toolbar) as Toolbar)
        supportActionBar!!.setDisplayHomeAsUpEnabled(true)
        imageView = findViewById<View>(R.id.imageView) as ImageView
        findViewById<View>(R.id.button_fade).setOnClickListener(onFadeButtonClick())
        findViewById<View>(R.id.button_slide_right).setOnClickListener(onSlideRightButtonClick())
        findViewById<View>(R.id.button_slide_left).setOnClickListener(onSlideLeftButtonClick())
        findViewById<View>(R.id.button_scale).setOnClickListener(onScaleButtonClick())
        findViewById<View>(R.id.button_zoom).setOnClickListener(onZoomButtonClick())
        newIntent = Intent(baseContext, TestActivity::class.java)
        newIntent!!.flags = Intent.FLAG_ACTIVITY_NEW_TASK
    }

    private fun onZoomButtonClick(): View.OnClickListener {
        return View.OnClickListener {
            val opts = ActivityOptionsCompat.makeCustomAnimation(baseContext,
                    R.anim.zoom_enter, R.anim.zoom_enter)
            ActivityCompat.startActivity(baseContext, newIntent!!, opts.toBundle())
        }
    }

    private fun onScaleButtonClick(): View.OnClickListener {
        return View.OnClickListener { v ->
            bundle = ActivityOptionsCompat.makeScaleUpAnimation(v, 0, 0, v.width, v.height).toBundle()
            ActivityCompat.startActivity(baseContext, newIntent!!, bundle)
        }
    }

    private fun onSlideLeftButtonClick(): View.OnClickListener {
        return View.OnClickListener {
            bundle = Bundle()
            bundle!!.putInt(Constants.KEY_ANIMATION, Constants.ANIMATION_SLIDE_LEFT)
            newIntent!!.putExtras(bundle!!)

            activityOptionsCompat = ActivityOptionsCompat.makeCustomAnimation(baseContext,
                    R.anim.slide_in_left, R.anim.slide_out_left)
            ActivityCompat.startActivity(baseContext, newIntent!!, activityOptionsCompat!!.toBundle())
        }
    }

    private fun onSlideRightButtonClick(): View.OnClickListener {
        return View.OnClickListener {
            bundle = Bundle()
            bundle!!.putInt(Constants.KEY_ANIMATION, Constants.ANIMATION_SLIDE_RIGHT)
            newIntent!!.putExtras(bundle!!)

            activityOptionsCompat = ActivityOptionsCompat.makeCustomAnimation(baseContext,
                    R.anim.slide_in_right, R.anim.slide_out_right)
            ActivityCompat.startActivity(baseContext, newIntent!!, activityOptionsCompat!!.toBundle())
        }
    }

    private fun onFadeButtonClick(): View.OnClickListener {
        return View.OnClickListener {
            bundle = Bundle()
            bundle!!.putInt(Constants.KEY_ANIMATION, Constants.ANIMATION_FADE)
            newIntent!!.putExtras(bundle!!)

            activityOptionsCompat = ActivityOptionsCompat.makeCustomAnimation(baseContext,
                    android.R.anim.fade_in, android.R.anim.fade_out)
            ActivityCompat.startActivity(baseContext, newIntent!!, activityOptionsCompat!!.toBundle())
        }
    }
}
